pkg/controller_helper/model_source/modelhub.go

```diff
@@ -89,6 +91,16 @@ func (p *ModelHubProvider) InjectModelLoader(template *corev1.PodTemplateSpec, i
 			corev1.EnvVar{Name: "REVISION", Value: *p.modelRevision},
 		)
 	}
+	if p.modelAllowPatterns != nil {
+		initContainer.Env = append(initContainer.Env,
+			corev1.EnvVar{Name: "MODEL_ALLOW_PATTERNS", Value: *p.modelAllowPatterns},
+		)
+	}
+	if p.modelIgnorePatterns != nil {
+		initContainer.Env = append(initContainer.Env,
+			corev1.EnvVar{Name: "MODEL_IGNORE_PATTERNS", Value: *p.modelIgnorePatterns},
```

Comment:
Doesn't this a **String** value, but we want a **Optional[List[str]]** in python side, will this work?

I just want to highlight that we hardcoded the loader image name in `pkg/defaults.go`, so what we changed in python side will only be tested part of them:
- what we have tested is the refactors in controller side is correct
- what we haven't tested is the new added patterns

So you may have to do the extra steps to make sure the code is working:
- make loader-image-load -> to build the right loader image
- replace the image name in `pkg/defaults.go`
- run the e2e tests(you may need to run `kind load image` to load the image since you didn't push to the remote registry)

I'm sorry for the trivial steps, but I have't found better ways right now.